Alternative Windows license management module

GrimAnEye opened the pull request:
### What does this PR do?
The existing Windows and MS Office license management module is based on supplied vbs-scripts from these products themselves.

Microsoft itself is [phasing out vbscript](https://learn.microsoft.com/en-us/windows/whats-new/deprecated-features-resources#vbscript), but what remains is a flexible and functional CIM( [1](https://learn.microsoft.com/en-us/previous-versions/windows/desktop/sppwmi/software-license-provider-), [2](https://learn.microsoft.com/ru-ru/powershell/module/cimcmdlets/invoke-cimmethod?view=powershell-7.4) ) tool that can be called via Powershell.

My module is an implementation of license key management via CIM methods, universal to all software that uses the Windows internal license store.

### What issues does this PR fix or reference?
None, it's a new module

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**
<!-- Please review the [test documentation](https://docs.saltproject.io/en/master/topics/tutorials/writing_tests.html) for details on how to implement tests into Salt's test suite. -->
- [ ] Docs - _there are comments on the code_
- [ ] Tests written/updated - _no tests, need help implementing them_

### Commits signed with GPG?
No

GrimAnEye committed:
new module

GrimAnEye committed:
update module and state

GrimAnEye committed:
update license

GrimAnEye committed:
upd module and state

GrimAnEye committed:
comments fix, "present" is completed, add more checks

GrimAnEye committed:
fix state output

GrimAnEye committed:
fix formatting

GrimAnEye committed:
add description

bdrx312 commented on salt/modules/win_mslicense.py:
```diff
@@ -0,0 +1,580 @@
+        salt '*' mslicense.install XXXXX-XXXXX-XXXXX-XXXXX-XXXXX
+    """
+
+    cmd = "Invoke-CimMethod -Query 'Select * From SoftwareLicensingService' -MethodName InstallProductKey -Arguments @{{ProductKey='{0}'}}".format(
```
I recommend adding escaping and/or validation to any arguments that are extrapolated directly as a string to a powershell command. The single quotes `'` you have help with most of it so I think the only character you have to worry about/deal with is a single quote; so, any single quotes need to be escaped which in powershell is with double single quotes `''`. So `key.replace("'", "''")`.

bdrx312 reviewed (dismissed):
Thanks for this contribution. There is a now a requirement to have tests for any pull request to add pytests, can you add some tests?

twangboy commented on salt/states/win_mslicense.py:
```diff
@@ -0,0 +1,541 @@
+
+    if __opts__["test"]:
+        ret["result"] = None
+        ret["comment"] = "{0} would be present".format(name[-5:])
```
Key ending with XXXXX will be activated

twangboy commented on salt/states/win_mslicense.py:
```diff
@@ -0,0 +1,541 @@
+
+    ret = {"name": name[-5:], "result": None, "comment": "", "changes": {}}
+
+    def infoIsNone(t: tuple, k: str): return None if (
```
We don't use camelCase in Salt unless we're working with classes. You should probably make this a hidden function as well by preceding it with an underscore (`_`) so it is not exposed in the documentation. Please use `_info_is_none`.

twangboy reviewed (dismissed):
This will need a change log and tests written.

With the suggested changes to the execution module, there will be substantial changes to the state module so they work together again.

Please reference the following execution module for the preferred docs format for Salt:
https://github.com/saltstack/salt/blob/master/salt/modules/win_wusa.py

twangboy committed:
Rewrite win_license module using PowerShell CIM

Replace slmgr.vbs-based implementation with PowerShell CIM via
SoftwareLicensingService/SoftwareLicensingProduct. Supports managing
any Windows Software Licensing Service product, not just Windows OS.

Add KMS management functions: get/set/clear host and port, with
registry-based port query to correctly distinguish unconfigured from
default (1688).

Remove win_mslicense module (superseded).
Add functional tests for license info and KMS round-trip operations.

Fixes #67043

dwoz commented:
@twangboy Please give this one a good review, otherwise I think it's ready.
